README.md

```diff
@@ -46,86 +46,102 @@ Take a few minutes to familiarize yourself with the OpenShift console. If you us
 
 In the OpenShift console, you can see the different deployments for the various EnMasse components. You can go into each pod and look at the logs. If we go to the address controller log, you can see that its creating a 'default' address space.
 
-#### Authenticating
-
-Go to the OpenShift console, application -> routes, and click on the hostname for the 'keycloak' route. This should bring you to the keycloak admin console. The admin user is protected by an automatically generated password, so we need to extract that as well before being able to create users.
+#### Authentication and Authorization
 
-```
-oc extract secret/keycloak-credentials -n $NAMESPACE
-cat admin.username
-cat admin.password
-```
+Go to the OpenShift console, application -> routes, and click on the hostname for the 'keycloak' route. This should bring you to the keycloak admin console. The admin user is protected by the password that was set in the playbook.
 
-In the Keycloak UI, create a new user, and a set of credentials for that user. Make sure the user is
+In the Keycloak UI, make sure you are in the 'default' realm. Then create a new user, and a set of credentials for that user. Make sure the user is
 enabled, and that the credentials are not marked as temporary.
 
 For this workshop we could use following users for example :
 
 * _console_ : as a "real" user who uses the console for creating addresses
-* _deviceX_ : as deviceX (i.e. device1, device2, ...) user
-* _sparkdriver_ : as Spark driver application user
+* _deviceX_ : as deviceX (i.e. device1, device2, ...) user who can only send/recv from a particular address
+* _sparkdriver_ : as Spark driver application user 
 * _thermostat_ : as thermostat application user
 
-It's clear that this workshop involves a "real" user who is in charge to create and configure the addresses and some other users which are relaled to applications running in the cluster and devices running in the field.
+It's clear that this workshop involves a "real" user who is in charge to create and configure the addresses and some other users which are related to applications running in the cluster and devices running in the field.
+
+It is now time to create some authorization rules to ensure the different users can only access the
+addresses assigned. This is done by creating groups with names on the form `send_myqueue` and have a
+user join that group to gain the permission (to send messages to address `myqueue`). We can for
+example create the following groups that will match the addresses that we will create later:
+
+* manage
+* send\_temperature
+* recv\_temperature
+* send\_max
+* recv\_max
+* send\_control
+* recv\_control
+
+To enable authorization, we need to have the users join the appropriate groups. For this workshop we
+can use the following mapping:
+
+* console - manage
+* deviceX - send\_temperature, recv\_control
+* sparkdriver - recv\_temperature, send\_max
+* thermostat - recv\_max, send\_control
```

Comment:
as for devices, the thermostat needs the rights to send to each `control\deviceX` address. Does the permission just on `control` work in a hierarchical way ?